kratos/geometries/brep_point.h

```diff
@@ -0,0 +1,352 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+
+#if !defined(KRATOS_BREP_POINT_H_INCLUDED )
+#define  KRATOS_BREP_POINT_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "geometries/geometry.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/**
+ * @class BrepSurface
+ * @ingroup KratosCore
+ * @brief The BrepSurface acts as topology for faces. Those
+ *        can be enclosed by a certain set of brep face curves.
+ */
+template<class TContainerPointType, int TLocalSpaceDimension, int TWorkingSpaceDimension>
```

Comment:
Why we need `TWorkingSpaceDimension`?

Reply:
It is actually possible to use this type of geometry for any working space dimension. Accordingly, it facilitates to keep the correct working space dimension within this geometry.